### PR TITLE
[#8316] [Platform] [API] Backups controller error handling.

### PR DESCRIPTION
**Pull request by jitendra-12113**
**Description:**
Use the new global exception handling in the Backups controller.

**Test Plan:**

Its mostly refactor. Unit tests are updated.
Run server and test basic operations.

**TODO:** Will work on followup change to get rid of raw json used in this controller 
#8315

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/controllers/BackupsController.java [hunk 1]
```diff
@@ -160,11 +152,7 @@ public Result restore(UUID customerUUID, UUID universeUUID) {
   }
 
   public Result delete(UUID customerUUID) {
-    Customer customer = Customer.get(customerUUID);
-    if (customer == null) {
-      String errMsg = "Invalid Customer UUID: " + customerUUID;
-      return ApiResponse.error(BAD_REQUEST, errMsg);
-    }
+    Customer customer = Customer.getOrBadRequest(customerUUID);
     ObjectNode formData = (ObjectNode) request().body().asJson();
```
[the remark this answers is not in the document]
Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/controllers/BackupsController.java:156]: Actually you can do this in followup diff. Just add a TODO(API)

File: managed/src/main/java/com/yugabyte/yw/models/CustomerConfig.java [hunk 2]
```diff
@@ -135,6 +137,15 @@ public static CustomerConfig get(UUID customerUUID, UUID configUUID) {
...
 
+  public static CustomerConfig getOrBadRequest(UUID customerUUID, UUID configUUID) {
+    CustomerConfig storageConfig = get(customerUUID, configUUID);
+    if (storageConfig == null) {
+      throw new YWServiceException(BAD_REQUEST, "Invalid StorageConfig UUID: "
+          + configUUID);
+    }
+    return storageConfig;
+  }
+
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/models/CustomerConfig.java:148]: Mark get method @Deprecated so people are encouraged to use getOrBadRequest

**Review by sb-yb — APPROVED** [review]
Just minor comments.

**Review by mahendranbhat — APPROVED** [review]

File: managed/src/main/java/com/yugabyte/yw/controllers/BackupsController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/BackupsController.java:156]: Created #8315  to track raw json cleanup. Thanks!

File: managed/src/main/java/com/yugabyte/yw/models/CustomerConfig.java (hunk 2 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/models/CustomerConfig.java:148]: Done.

**Pull request merged**